<launch>
  <arg name="average_point_cloud_topic_name" default="" />
  <arg name="min_max_hsi_topic_name" default="" />
  <arg name="extract_point_cloud_topic_name" default="" />
  <arg name="input_cloud" default="" />
  <arg name="input_image" default="" />
  <arg name="manager" default="" />
  <arg name="identify_name" default="" />

  <!-- attention_clipper -->
  <include file="$(find wrapping_melodic)/launch/hiro/attention_clipper.launch">
    <arg name="input_cloud" default="$(arg input_cloud)" />
    <arg name="identify_name" default="$(arg identify_name)" />
    <arg name="manager"  value="$(arg manager)" />
  </include>

  <!-- save pcd -->
  <arg name="input_cloud_pcd" value="/bbox_cloud_extract_$(arg identify_name)/output" />
  <node name="pointcloud_to_pcd_$(arg identify_name)" type="nodelet" pkg="nodelet"
        args="load jsk_pcl_utils/PointCloudToPCD $(arg manager)" output="screen" >
    <remap from="~input" to="$(arg input_cloud_pcd)" />
    <rosparam subst_value="true">
      prefix: $(find wrapping_melodic)/save_pcd/tape/
      duration: 0.0
      binary: false
      compressed: false
    </rosparam>
  </node>

  <!-- average pointcloud -->
  <node name="average_point_cloud_$(arg identify_name)" type="detect_tape" pkg="wrapping_melodic" output="screen" >
    <param name="extract_point_cloud_topic" value="$(arg extract_point_cloud_topic_name)"/>
    <param name="average_point_cloud_topic" value="$(arg average_point_cloud_topic_name)"/>
    <param name="min_max_hsi_topic" value="$(arg min_max_hsi_topic_name)"/>
  </node>
</launch>
